From 5ecf68ba193d69bb9fa2453fe7a65c0a2a926630 Mon Sep 17 00:00:00 2001
From: triallax <triallax@tutanota.com>
Date: Tue, 30 Apr 2024 10:22:33 +0100
Subject: [PATCH] settings: make `option_entries` NULL-terminated

As stated in the documentation of `gtk_init_with_args` (to which
`option_entries` is passed):

> The array must be NULL-terminated.

(source: https://docs.gtk.org/gtk3/func.init_with_args.html)

On Chimera Linux, this was causing `thunar-volman-settings` to segfault
when literally any argument was passed to it, whether an option or not.
---
https://gitlab.xfce.org/xfce/thunar-volman/-/merge_requests/18

 thunar-volman-settings/main.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/thunar-volman-settings/main.c b/thunar-volman-settings/main.c
index 53645f6..25c2903 100644
--- a/thunar-volman-settings/main.c
+++ b/thunar-volman-settings/main.c
@@ -42,6 +42,7 @@ static gint opt_socket_id = 0;
 GOptionEntry option_entries[] =
 {
   { "socket-id", 's', G_OPTION_FLAG_IN_MAIN, G_OPTION_ARG_INT, &opt_socket_id, N_("Settings manager socket"), N_("SOCKET ID") },
+  { NULL }
 };
 
 
-- 
GitLab

